tm2/pkg/sdk/baseapp.go

```diff
@@ -653,6 +653,7 @@ func (app *BaseApp) runMsgs(ctx Context, msgs []Msg, mode RunTxMode) (result Res
 		// each result.
 		data = append(data, msgResult.Data...)
 		events = append(events, msgResult.Events...)
+		events = append(events, ctx.EventLogger().Events()...)
```

Comment:
I think these should be appended in a defer statement, so that the event logs are always appended to the returned result, from handleMsgXXX.

Reply:
@notJoon, do you intend to implement this change? If not, could you please explain why not?

Reply:
now revised it like below:
```go
defer func() {
    events = append(events, ctx.EventLogger().Events()...)
    result.Events = events
}()
```